pkg/services/store/service.go

```diff
@@ -86,3 +108,82 @@ func (s *standardStorageService) Read(ctx context.Context, user *models.SignedIn
 	// TODO: permission check!
 	return s.tree.GetFile(ctx, path)
 }
+
+func isFileTypeValid(filetype string) bool {
+	if (filetype == "image/jpeg") || (filetype == "image/jpg") || (filetype == "image/gif") || (filetype == "image/png") || (filetype == "image/svg+xml") || (filetype == "image/webp") {
+		return true
+	}
+	return false
+}
+
+func (s *standardStorageService) Upload(ctx context.Context, user *models.SignedInUser, form *multipart.Form) (*Response, error) {
+	response := Response{
+		path:       "upload",
+		statusCode: 200,
+		message:    "Uploaded successfully",
+		err:        false,
+	}
+	upload, _ := s.tree.getRoot("upload")
+	if upload == nil {
+		response.statusCode = 404
+		response.message = "upload feature is not enabled"
+		response.err = true
+		return &response, fmt.Errorf("upload feature is not enabled")
+	}
+
+	files := form.File["file"]
+	for _, fileHeader := range files {
+		// Restrict the size of each uploaded file to 1MB.
+		if fileHeader.Size > MAX_UPLOAD_SIZE {
+			response.statusCode = 400
+			response.message = "The uploaded image is too big"
+			response.err = true
+			return &response, nil
+		}
+
+		// open each file to copy contents
+		file, err := fileHeader.Open()
+		if err != nil {
+			return nil, err
+		}
+		err = file.Close()
+		if err != nil {
+			return nil, err
+		}
+		data, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
+		filetype := http.DetectContentType(data)
+		// only allow images to be uploaded
+		if !isFileTypeValid(filetype) && !strings.HasSuffix(fileHeader.Filename, ".svg") {
+			return &Response{
+				statusCode: 400,
+				message:    "unsupported file type uploaded",
+				err:        true,
+			}, nil
+		}
+		err = upload.Upsert(ctx, &filestorage.UpsertFileCommand{
+			Path:     "/" + fileHeader.Filename,
```

Comment:
Should we protect file from being overwritten by a file with the same name uploaded by an another user in a different panel?

Reply:
Could this be a UX thing? Like have a confirmation dialogue that says "file with the same name already exists in _x_ folder, override or change file name?"